agent/command/assume_ec2_role.go

```diff
@@ -71,6 +70,21 @@ func (r *ec2AssumeRole) Execute(ctx context.Context,
 		return errors.WithStack(err)
 	}
 
+	if r.TemporaryFeatureFlag {
+		return r.newExecution(ctx, comm, logger, conf)
+	}
+
+	return r.legacyExecute(ctx, comm, logger, conf)
+}
+
+func (r *ec2AssumeRole) newExecution(ctx context.Context,
```

Comment:
nit: let's name this execute since the other function already specifies that it is legacy